packages/components-react/projects/react-ssr-wrapper/tests/unit/specs/splitChildren.spec.ts

```diff
@@ -0,0 +1,55 @@
+import type { ReactElement } from 'react';
+import { splitChildren } from '../../../src/splitChildren';
+
+const textEl = 'Some text' as unknown as ReactElement;
+
+const reactEl = {
+  type: 'br',
+  key: null,
+  ref: null,
+  props: {},
+  _owner: null,
+  _store: {},
+};
+
+const reactElWithSlot = {
+  type: 'span',
+  key: null,
+  ref: null,
+  props: {
+    slot: 'heading',
+    style: [Object],
+    children: 'Some heading',
+  },
+  _owner: null,
+  _store: {},
+};
+
+describe('splitChildren()', () => {
+  it.each<[ReactElement | ReactElement[] | undefined, ReactElement[]]>([
```

Comment:
Very strange setup. 
Considering using the correct types and what the function actually returns the entire output of the function should be tested, not just parts of it.  
Therefore, the the `it.each` should be typed something like
```
it.each<[ReactNode, { children: ReactNode[], otherChildren: ReactNode[], namedSlotChildren: ReactNode[]}]>
```